### PR TITLE
Detect RPC error responses for settile call

### PR DESCRIPTION
**Pull request by ghost**
Adds detection of RPC error responses for the `settile` call in the publish procedures. Eliminates a chunk of silent publication failures.

Cause of a publication failure is output to the Unity console.

Changes the structure of the `RPCResponse` class to match the data returned from the node. I find no uses of this class outside the publish procedure. Don't believe the change will hurt anything.

In tests of this change I have:
1. Successfully published and detected the content with `gettile`.
1. Seen descriptive errors for failures that were previously silent.

### CODE REVIEW

File: Assets/Source/LandGenerator.cs [hunk 1]
```diff
@@ -202,9 +202,15 @@ public class LandGenerator : MonoBehaviour {
...
+public class RPCError {
+  public string message;
+  public int code;
+}
+
+[System.Serializable]
 public class RPCResponse {
 	public string result = null;
-	public string error = null;
+  public RPCError error = null;
```
Comment (martriay) [inline Assets/Source/LandGenerator.cs:213]: be careful with the indentation!

File: Assets/Editor/DecentralandEditor.cs [hunk 2]
```diff
@@ -147,14 +147,38 @@ public class DecentralandEditor : EditorWindow {
...
-		if (string.IsNullOrEmpty(www.error)) {
-			RPCResponse response = JsonUtility.FromJson<RPCResponse>(www.text);
-			Debug.Log("Successfully published tile!");
-			published = true;
-		} else {
-			Debug.Log("Error publishing tile! " + www.error);
-			publishError = true;
-		}
+    // Web transaction error
+    if(!string.IsNullOrEmpty(www.error)) {
```
Comment (martriay) [inline Assets/Editor/DecentralandEditor.cs:151]: Please be consistent with the spacing (`if(` vs `if (`)

File: Assets/Source/LandGenerator.cs (hunk 1 above)
Comment (ghost) [reply Assets/Source/LandGenerator.cs:213]: Sorry about it, i noticed it too after I checked in. My text editor normalized and it didn't show this way for me. I will watch it.

File: Assets/Editor/DecentralandEditor.cs (hunk 2 above)
Comment (ghost) [reply Assets/Editor/DecentralandEditor.cs:151]: Agreed, I'd better be consistent. If a convention is established I'll follow it.

Comment (martriay) [reply Assets/Editor/DecentralandEditor.cs:151]: As below, i'd go with `if (`

**Review by eordano — APPROVED** [review]
Let's merge this and I'll fix the minor concerns with spacing

**Comment by ghost** [comment]
Thanks @eordano for picking it up. Sorry about my poor formatting. Still need to get my dev environment set up right to prevent this.

**Comment by martriay** [comment]
I've handled the spacing issues.

**Pull request merged**